packages/api/repository/ApplicationRepository.ts

```diff
@@ -0,0 +1,64 @@
+import { EntityRepository, Repository, InsertResult, UpdateResult } from "typeorm";
+import { Application, IApplicationCreate, IApplicationUpdate } from "../../orm/entity/Application";
+
+@EntityRepository(Application)
+export class ApplicationRepository extends Repository<Application> {
+  // GET all application names that match the provided name
+  getAllMatchingApplicationNames(portfolioId: string, appName: string): Promise<Array<Application>> {
+    return this.createQueryBuilder("application")
+      .select(["application.name"])
+      .where("application.portfolioId = :portfolioId", { portfolioId })
+      .andWhere("application.name = :appName", { appName })
+      .getMany();
+  }
```

Comment:
Nice! I think something like this crossed my mind, but I didn't actually pursue it since I had something working, but good refactor material.